src/code/utils/translate.js

```diff
@@ -10,24 +10,28 @@
 import urlParams  from './url-params'
 import de from './lang/de'
 import el from './lang/el'
-import enUs from './lang/en-US'
+import enUS from './lang/en-US'
 import es from './lang/es'
 import he from './lang/he'
+import ja from './lang/ja'
 import nb from './lang/nb'
 import nn from './lang/nn'
 import tr from './lang/tr'
-import zh from './lang/zh-TW'
+import zhHans from './lang/zh-Hans'
+import zhTW from './lang/zh-TW'
 
 const languageFiles = [
-  {key: 'de',    contents: de},  // German
-  {key: 'el',    contents: el},  // Greek
-  {key: 'en-US', contents: enUs},// US English
-  {key: 'es',    contents: es},  // Spanish
-  {key: 'he',    contents: he},  // Hebrew
-  {key: 'nb',    contents: nb},  // Norwegian Bokmål
-  {key: 'nn',    contents: nn},  // Norwegian Nynorsk
-  {key: 'tr',    contents: tr},  // Turkish
-  {key: 'zh-TW', contents: zh}   // Chinese (Taiwan)
+  {key: 'de',    contents: de},     // German
+  {key: 'el',    contents: el},     // Greek
+  {key: 'en-US', contents: enUS},   // US English
+  {key: 'es',    contents: es},     // Spanish
+  {key: 'he',    contents: he},     // Hebrew
+  {key: 'ja' ,   contents: ja},     // Japanese
+  {key: 'nb',    contents: nb},     // Norwegian Bokmål
+  {key: 'nn',    contents: nn},     // Norwegian Nynorsk
+  {key: 'tr',    contents: tr},     // Turkish
+  {key: 'zh-tw', contents: zhHans}, // Traditional Chinese (Taiwan)
```

Comment:
Due to brower's not returning zh-hans for simplified Chinese but returning zh-tw for traditional we defaulted to the following in sagemodeler (and I thought we had done the same in CFM):

```
  "zh":      require("./lang/zh-HANS.json"), // Chinese (Simplified)
  "zh-TW":   require("./lang/zh-TW.json"),   // Chinese (Traditional)
```

(https://github.com/concord-consortium/sage-modeler-site/blob/master/src/code/utils/translate.ts#L12-L13)

this lets us default to simplfied Chinese for all zh readers except for those that exactly specify zh-tw.

More info in these two threads on Slack:

- https://concord-consortium.slack.com/archives/C09JNV371/p1591364967116300
- https://concord-consortium.slack.com/archives/C09JNV371/p1591372571128300

Reply:
Thx for the pointers -- I'll incorporate what I can from those links and let you know when it's ready for re-review.